Calibration/IsolatedParticles/src/CaloPropagateTrack.cc

```diff
@@ -150,7 +169,10 @@ namespace spr{
 	if (std::abs(point.eta())<spr::etaBEEcal) {
 	  vdet.detIdECAL = barrelGeom->getClosestCell(point);
 	} else {
-	  vdet.detIdECAL = endcapGeom->getClosestCell(point);
+	  if (endcapGeom) 
```

Comment:
should be endcapGeom != null_ptr
@bsunanda 
@smuzaffar - do you see why our code checks do not catch this? (I think they should..)

Reply:
@davidlange6 isn't that a matter of taste?
I seem to remember that even Bjarne Stroustrup preferred the implicit cast because it conveys better the question "if the object is valid".

Reply:
I am not sure what I should do now - should we try to see if it is null_ptr or the current one is fine